web/src/settings_components.ts

```diff
@@ -887,13 +850,95 @@ export function check_property_changed(
                 "#org-notifications .language_selection_widget .language_selection_button span",
             ).attr("data-language-code");
             break;
+        default:
+            if (current_val !== undefined) {
+                proposed_val = get_input_element_value(elem, typeof current_val);
+            } else {
+                blueslip.error("Element refers to unknown property", {property_name});
+            }
+    }
+    return current_val !== proposed_val;
+}
+
+export function check_stream_settings_property_changed(
+    elem: HTMLElement,
+    sub: StreamSubscription,
+): boolean {
+    const $elem = $(elem);
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const property_name = extract_property_name($elem) as StreamSettingProperties;
+    const current_val = get_stream_settings_property_value(property_name, sub);
+    let proposed_val;
+    switch (property_name) {
+        case "can_remove_subscribers_group":
+            proposed_val = get_dropdown_list_widget_setting_value($elem);
+            break;
+        case "message_retention_days":
+            assert(elem instanceof HTMLSelectElement);
+            proposed_val = get_message_retention_setting_value($(elem), false);
+            break;
+        case "stream_privacy":
+            proposed_val = get_input_element_value(elem, "radio-group");
+            break;
+        default:
+            if (current_val !== undefined) {
+                proposed_val = get_input_element_value(elem, typeof current_val);
+            } else {
+                blueslip.error("Element refers to unknown property", {property_name});
+            }
+    }
+    return current_val !== proposed_val;
+}
+
+export function check_group_property_changed(elem: HTMLElement, group: UserGroup): boolean {
+    const $elem = $(elem);
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const property_name = extract_property_name($elem) as keyof UserGroup;
+    const current_val = get_group_property_value(property_name, group);
+    let proposed_val;
+    if (property_name === "can_mention_group") {
+        proposed_val = get_dropdown_list_widget_setting_value($elem);
+    } else if (current_val !== undefined) {
+        proposed_val = get_input_element_value(elem, typeof current_val);
```

Comment:
Can we update this to use `switch` like other functions do? We would soon add new settings so would be good to just write the code considering that.